Algorithm/QCAlgorithm.Trading.cs

```diff
@@ -937,7 +937,16 @@ private OrderResponse PreOrderChecksImpl(SubmitOrderRequest request)
 
             if (request.OrderType == OrderType.MarketOnClose)
             {
-                var nextMarketClose = security.Exchange.Hours.GetNextMarketClose(security.LocalTime, false);
+                DateTime nextMarketClose;
+
+                if (security.Exchange.Hours.IsMarketAlwaysOpen)
+                {
+                    nextMarketClose = security.LocalTime + new TimeSpan(24, 0, 0);
```

Comment:
Think we should throw here, it wont ever happen, a `MarketOnClose` order in an asset which has no market close, same comment for the `FillModel`. We should have the same for `MarketOnOpen` 👍